### PR TITLE
Will now create a pull request in app-sre with the versions JSON

### PR DESCRIPTION
**Pull request by omertuc**
@oshercc

### CODE REVIEW

**Review by oshercc — APPROVED** [review]

File: tools/new_minor_OCP_version.py [hunk 1]
```diff
@@ -51,12 +64,18 @@ def main(args):
...
-        task = create_task(args, current_assisted_service_ocp_version, latest_ocp_version)
+        ticket_id = create_task(args, current_assisted_service_ocp_version, latest_ocp_version)
+
         # task is not created in case task is available
-        if task:
-            branch = update_ai_repo_to_new_ocp_version(args, current_assisted_service_ocp_version, latest_ocp_version, task)
-            pr = open_pr(args, current_assisted_service_ocp_version, latest_ocp_version , task)
-            test_changes(branch, pr)
+        if ticket_id is not None:
+            branch, openshift_versions_json = update_ai_repo_to_new_ocp_version(args, current_assisted_service_ocp_version, latest_ocp_version, ticket_id)
```
Comment (eranco74) [inline tools/new_minor_OCP_version.py:71]: Add logging

File: tools/new_minor_OCP_version.py [hunk 2]
```diff
@@ -184,6 +205,16 @@ def clone_assisted_service(user_password):
     cmd = ASSISTED_SERVICE_CLONE_CMD.format(user_password=user_password)
     subprocess.check_output(cmd, shell=True)
 
+def clone_app_interface(gitlab_key):
+    logging.info(DELETE_APP_INTERFACE_CLONE_CMD)
+    subprocess.check_output(DELETE_APP_INTERFACE_CLONE_CMD, shell=True)
```
Comment (eranco74) [inline tools/new_minor_OCP_version.py:210]: If you don't care about the output use check_call

Comment (eranco74) [inline tools/new_minor_OCP_version.py:210]: Avoid `shell=true`

File: tools/new_minor_OCP_version.py [hunk 3]
```diff
@@ -196,6 +227,39 @@ def change_version_in_files(old_version ,new_version):
                 file=file
             ),shell=True)
 
+def add_single_node_fake_4_8_release_image(openshift_versions_json):
+    versions = json.loads(openshift_versions_json)
```
Comment (eranco74) [inline tools/new_minor_OCP_version.py:231]: Consider moving this to another file ( e.g. non_production_release.images.json) and get the content from there

File: tools/new_minor_OCP_version.py [hunk 4]
```diff
@@ -209,21 +273,72 @@ def commit_and_push_version_update_changes(new_version, message_prefix):
     subprocess.check_output("git push origin HEAD:{}".format(branch), cwd="assisted-service", shell=True)
     return branch
 
+def commit_and_push_version_update_changes_app_interface(gitlab_key, fork, new_version, message_prefix):
+    def cmd(*args, **kwargs):
```
Comment (eranco74) [inline tools/new_minor_OCP_version.py:277]: You seem to run a similar command in `clone_app_interface` consider moving this function out.

File: tools/new_minor_OCP_version.py (hunk 1 above)
Comment (omertuc) [reply tools/new_minor_OCP_version.py:71]: Is this comment relevant for this particular line, or the file in the general?

Comment (eranco74) [reply tools/new_minor_OCP_version.py:71]: Relevant for this if, add logging about the ticket you are working on

File: tools/new_minor_OCP_version.py (hunk 3 above)
Comment (omertuc) [reply tools/new_minor_OCP_version.py:237]: Solved in 1af5150

File: tools/new_minor_OCP_version.py [hunk 5]
```diff
@@ -17,232 +24,452 @@
...
-    jkusername, jkpassword = get_login(args.jenkins_user_password)
-    j = jenkins.Jenkins(JENKINS_URL, username=jkusername, password=jkpassword)
-    next_build_number = j.get_job_info(TEST_INFRA_JOB)['nextBuildNumber']
-    j.build_job(TEST_INFRA_JOB, parameters={"SERVICE_BRANCH": branch, "NOTIFY":False, "JOB_NAME": "Update_ocp_version"})
+
+def test_test_infra_passes(args, branch):
+    username, password = get_login(args.jenkins_user_password)
+    jenkins_client = jenkins.Jenkins(JENKINS_URL, username=username, password=password)
+    next_build_number = jenkins_client.get_job_info(TEST_INFRA_JOB)['nextBuildNumber']
+    jenkins_client.build_job(TEST_INFRA_JOB,
```
Comment (eranco74) [inline tools/new_minor_OCP_version.py:173]: This seems a bit racy, can't we get the build number from the build_job return value?

File: tools/new_minor_OCP_version.py [hunk 6]
```diff
@@ -17,232 +24,452 @@
...
+                             parameters={"SERVICE_BRANCH": branch, "NOTIFY": False, "JOB_NAME": "Update_ocp_version"})
     time.sleep(10)
-    while not j.get_build_info(TEST_INFRA_JOB, next_build_number)["result"]:
+    while not jenkins_client.get_build_info(TEST_INFRA_JOB, next_build_number)["result"]:
```
Comment (eranco74) [inline tools/new_minor_OCP_version.py:176]: Consider extracting this to `wait_for_job_to_finish` function

**Review by eranco74 — COMMENTED** [review]
lgtm
I'll open a jira epic for improving this repo.
it's scary to have all this code without a single test.
Also, we should probably have Jenkins and Jira modules that expose the required functionality instead of implementing everything in this file.

File: tools/new_minor_OCP_version.py (hunk 5 above)
Comment (omertuc) [reply tools/new_minor_OCP_version.py:173]: Not sure, not my code - I'm just refactoring. Note that it's sleeping if the job is not found.

File: tools/new_minor_OCP_version.py (hunk 6 above)
Comment (omertuc) [reply tools/new_minor_OCP_version.py:176]: https://github.com/eranco74/assisted-installer-deployment/pull/41/files#r571624504

**Comment by omertuc** [comment]
> lgtm
> I'll open a jira epic for improving this repo.
> it's scary to have all this code without a single test.
> Also, we should probably have Jenkins and Jira modules that expose the required functionality instead of implementing everything in this file.

It's not that scary because the code has a very low impact - it only creates tickets and PRs. Humans have to review and merge whatever this code does. For now, at-least

**Comment by eranco74** [comment]
> > lgtm
> > I'll open a jira epic for improving this repo.
> > it's scary to have all this code without a single test.
> > Also, we should probably have Jenkins and Jira modules that expose the required functionality instead of implementing everything in this file.
> 
> It's not that scary because the code has a very low impact - it only creates tickets and PRs. Humans have to review and merge whatever this code does. For now, at-least

Never trust humans :wink: 
The code might fail to notice changes or create the tickets.
It's not just this script, the entire repo is unmaintainable.

**Pull request merged**